src/scripts/comparison_of_horizontal_and_vertical_programs/analysis_maxHTM_scenario.py

```diff
@@ -0,0 +1,229 @@
+
+"""
+
+import datetime
+from pathlib import Path
+
+from tlo import Date, logging
+from tlo.methods import (
+    demography,
+    enhanced_lifestyle,
+    epi,
+    healthburden,
+    healthseekingbehaviour,
+    healthsystem,
+    hiv,
+    malaria,
+    simplified_births,
+    symptommanager,
+    tb,
+)
+from tlo.scenario import BaseScenario
+
+resourcefilepath = Path("./resources")
+datestamp = datetime.date.today().strftime("__%Y_%m_%d")
+
+outputspath = Path("./outputs")
+scaleup_start_year = 2012
+end_date = Date(2015, 1, 1)
+
+
+class EffectOfProgrammes(BaseScenario):
+    def __init__(self):
+        super().__init__()
+        self.seed = 0
+        self.start_date = Date(2010, 1, 1)
+        self.end_date = end_date
+        self.pop_size = 1_000
+        self.number_of_draws = 5
+        self.runs_per_draw = 1
+
+    def log_configuration(self):
+        return {
+            'filename': 'scaleup_tests',
+            'directory': Path('./outputs'),  # <- (specified only for local running)
+            'custom_levels': {
+                '*': logging.WARNING,
+                'tlo.methods.hiv': logging.INFO,
+                'tlo.methods.tb': logging.INFO,
+                'tlo.methods.malaria': logging.INFO,
+                'tlo.methods.demography': logging.INFO,
+            }
+        }
+
+    def modules(self):
+        return [
+            demography.Demography(resourcefilepath=self.resources),
+            simplified_births.SimplifiedBirths(resourcefilepath=self.resources),
+            enhanced_lifestyle.Lifestyle(resourcefilepath=self.resources),
+            healthsystem.HealthSystem(resourcefilepath=self.resources),
+            symptommanager.SymptomManager(resourcefilepath=self.resources),
+            healthseekingbehaviour.HealthSeekingBehaviour(resourcefilepath=self.resources),
+            healthburden.HealthBurden(resourcefilepath=self.resources),
+            epi.Epi(resourcefilepath=self.resources),
+            hiv.Hiv(resourcefilepath=self.resources),
+            tb.Tb(resourcefilepath=self.resources),
+            malaria.Malaria(resourcefilepath=self.resources),
+        ]
+
+    def draw_parameters(self, draw_number, rng):
+
+        return {
+            'Hiv': {
+                'type_of_scaleup': ['none', 'max', 'none', 'none', 'max'][draw_number],
+                'scaleup_start_year': scaleup_start_year,
+            },
+            'Tb': {
+                'type_of_scaleup': ['none', 'none', 'max', 'none', 'max'][draw_number],
+                'scaleup_start_year': scaleup_start_year,
+            },
+            'Malaria': {
+                'type_of_scaleup': ['none', 'none', 'none', 'max', 'max'][draw_number],
+                'scaleup_start_year': scaleup_start_year,
+            },
+        }
+
+
+if __name__ == '__main__':
+    from tlo.cli import scenario_run
+
+    scenario_run([__file__])
+
+
+
+# %% Produce some figures and summary info
```

Comment:
is this the code that generates the figures you showed me this morning?

Reply:
almost the same, edited to show just the 5 runs in the script (rather than mode1 vs mode2)